Userland/Applications/DisplaySettings/BackgroundSettingsWidget.cpp

```diff
@@ -100,16 +101,13 @@ ErrorOr<void> BackgroundSettingsWidget::create_frame()
 
     auto& button = *find_descendant_of_type_named<GUI::Button>("wallpaper_open_button");
     button.on_click = [this](auto) {
-        auto path_or_empty = GUI::FilePicker::get_open_filepath(window(), "Select wallpaper from file system", "/res/wallpapers"sv, false, GUI::Dialog::ScreenPosition::CenterWithinParent, { { GUI::FileTypeFilter::image_files(), GUI::FileTypeFilter::all_files() } });
-        if (!path_or_empty.has_value())
-            return;
-        auto path = String::from_deprecated_string(path_or_empty.value());
-        if (path.is_error()) {
+        auto response = FileSystemAccessClient::Client::the().open_file(window(), "Select wallpaper"sv, "/res/wallpapers"sv, Core::File::OpenMode::Read, { { GUI::FileTypeFilter::image_files(), GUI::FileTypeFilter::all_files() } });
+        if (response.is_error()) {
```

Comment:
Does this path not get traversed when cancel is pressed?

Reply:
It does, good catch!
We should definitely silently return in this case.

Reply:
Most likely. All other usages of `FileSystemAccessClient::Client::the().open_file` that I can see just return early.
```cpp
if (response.is_error()
    return;
```